### PR TITLE
Banner

### PR DESCRIPTION
**Pull request by alycrys**
- Remove `render` prop
- Remove `Link` component from content
- Create composable components `BannerContent` and `BannerCloseButton`

### CODE REVIEW

File: .changeset/olive-scissors-burn.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+"@salt-ds/lab": patch
+---
+
+Banner
```
Comment (libertymayc) [inline .changeset/olive-scissors-burn.md:5]: some other props added/removed need adding here

**Review by ZarkoPernar — DISMISSED** [review]

File: packages/lab/src/__tests__/__e2e__/banner/Banner.cy.tsx [hunk 2]
```diff
@@ -78,3 +45,39 @@ describe("GIVEN a Banner", () => {
     });
   });
 });
+
+describe("WHEN adding BannerCloseButton", () => {
+  beforeEach(() => {
```
Comment (joshwooding) [inline packages/lab/src/__tests__/__e2e__/banner/Banner.cy.tsx:50]: I would prefer to have a cy.mount per test

**Review by joshwooding — APPROVED** [review]

**Review by joshwooding — CHANGES_REQUESTED** [review]
See comment about the hook

File: packages/lab/src/banner/Banner.tsx [hunk 3]
```diff
@@ -115,67 +65,31 @@ export const Banner = forwardRef<HTMLDivElement, BannerProps>(function Banner(
...
-    return {
-      className: clsx(withBaseName("icon"), status, className),
-      ...restProps,
-    };
-  };
-
-  const getLabelProps = ({ className, ...restProps }: LabelProps = {}) => ({
-    className: clsx(withBaseName("label"), status, className),
-    ...restProps,
+  const [open] = useControlled({
```
[the remark this answers is not in the document]
Comment (alycrys) [reply packages/lab/src/banner/Banner.tsx:68]: true that. thanks

File: packages/lab/src/__tests__/__e2e__/banner/Banner.cy.tsx (hunk 2 above)
Comment (alycrys) [reply packages/lab/src/__tests__/__e2e__/banner/Banner.cy.tsx:50]: any particular reason ?

Comment (joshwooding) [reply packages/lab/src/__tests__/__e2e__/banner/Banner.cy.tsx:50]: It makes changing tests much easier and reduces having to check further up the file to find the mounted component. I think there are a lot of tests here that could be combined too.

File: packages/lab/src/banner/Banner.tsx [hunk 4]
```diff
@@ -1,100 +1,49 @@
...
-   * onClose callback when the close icon is clicked, the dismiss button will not be rendered if this is
-   * not defined
-   */
-  onClose?: (event: MouseEvent<HTMLButtonElement>) => void;
-  /**
-   * render props callback (if children callback not supplied)
+   * Controlled prop to set visible state.
    */
-  render?: (props: GetStateAndPropGetters) => ReactNode;
+  open?: boolean;
```
Comment (joshwooding) [inline packages/lab/src/banner/Banner.tsx:30]: Just a question but is it still worth having an open prop versus conditionally mounting the component?

Comment (alycrys) [reply packages/lab/src/banner/Banner.tsx:30]: Most probably we don't :)

File: packages/lab/src/banner/Banner.tsx [hunk 5]
```diff
@@ -115,67 +59,20 @@ export const Banner = forwardRef<HTMLDivElement, BannerProps>(function Banner(
...
-  }
-
   return (
     <div
       className={clsx(withBaseName(), withBaseName(status), className, {
         [withBaseName("emphasize")]: emphasize,
       })}
       ref={handleRef}
       {...rest}
+      aria-live="polite"
```
[the remark this answers is not in the document]
Comment (alycrys) [reply packages/lab/src/banner/Banner.tsx:69]: if it's not needed, I'll take it out in my next PR

**Pull request merged**